packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1871,49 +1883,47 @@ function updateEffect(
   return updateEffectImpl(PassiveEffect, HookPassive, create, deps);
 }
 
+function useEventImpl<T>(event: EventFunction<T>, nextEvent: () => T) {
+  const eventState = {event, nextEvent};
```

Comment:
Another trick we sometimes use is to push each of these items into the array separately. Then when iterating over the array, you know the ith values are the wrapper functions, and the (i + 1)th values are the new impls. I believe the array won't deopt because both of these values are functions.